python/llm/src/bigdl/llm/transformers/model.py

```diff
@@ -29,63 +29,80 @@ class _BaseAutoModelClass:
+            subfolder = kwargs.get("subfolder", "")
+            variant = kwargs.get("variant", None)
+            qtype = config_dict.pop("bigdl_transformers_dtype", 'q4_0')
 
-        model = cls.HF_Model.from_pretrained(*args, **kwargs)
-        print("Note: If there are warnings about mismatched during the loading process, "
-              "please ignore them as it is part of the normal flow. "
-              "The model will be reconverted to the format of BigDL after loading.")
-
-        # Note that the ggml_matmul_src1_x_src0_t operation cannot currently
-        # be recorded in AutoConfig,
-        # and this operation is not included in the core Hugging Face infrastructure.
-        if bigdl_transformers_int4:
             from .convert import ggml_convert_quant
+            model = cls.HF_Model.from_pretrained(*args, **kwargs)
+            print("Note: If there are warnings during the model loading process, "
+                  "they can be safely ignored; "
+                  "the model will be loaded with INT4 optimizations applied.")
+
             # We forcefully modify the model's definition
             # and the tensor shape of int4 weights without quantization.
-            model = ggml_convert_quant(model, convert_shape_only=True)
+            model = ggml_convert_quant(model, qtype, convert_shape_only=True)
             # Load the quantized model at last.
             archive_file = extract_local_archive_file(pretrained_model_name_or_path,
                                                       subfolder,
                                                       variant)
             state_dict = load_state_dict(archive_file)
             load(model, state_dict)
             del state_dict
-        elif qtype:
-            from .convert import ggml_convert_quant
-            model = model.to("cpu")
-            model = ggml_convert_quant(model, qtype)
-            model.config.update({"bigdl_transformers_int4": True})
 
+        elif load_in_4bit:
+            qtype = ggml_tensor_qtype['q4_0']
+            model = cls.convert_quant(model, qtype, *args, **kwargs)
+        elif load_in_low_bit:
+            load_in_low_bit = load_in_low_bit.lower()
+            invalidInputError(load_in_low_bit in ggml_tensor_qtype,
+                              f"Unknown load_in_low_bit value: {load_in_low_bit},"
+                              f" excepted q4_0, q4_1, q5_0, q5_1, q8_0.")
+            qtype = ggml_tensor_qtype[load_in_low_bit]
+            model = cls.convert_quant(model, qtype, *args, **kwargs)
+
+        return model
+
+    @classmethod
+    def convert_quant(cls, model, qtype, *args, **kwargs):
+        from .convert import ggml_convert_quant
+        model = cls.HF_Model.from_pretrained(*args, **kwargs)
+        model = model.to("cpu")
+        model = ggml_convert_quant(model, qtype)
+        model.config.update({"bigdl_transformers_low_bit": True,
+                             "bigdl_transformers_dtype": qtype})
```

Comment:
Maybe you just need only  one parameter "bigdl_transformers_low_bit": qtype

Reply:
yes it is

Reply:
You may combine `bigdl_transformers_low_bit` and `bigdl_transformers_dtype`

Reply:
fixed